### ISSUE TITLE
Cannot build the project: Givng tons of errors!

### ISSUE THREAD

**Issue by anuragvohraec** [kriszyp/lmdbx-store#1]
When I run npm install it give me long lits of errors.

```
yp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/anurag/ws/db/lmdbx-store/build'
  CC(target) Release/obj.target/lmdb-store/dependencies/libmdbx/src/core.o
../dependencies/libmdbx/src/core.c:22189:2: warning: #warning "Build flags undefined. Please use correct build script" [-Wcpp]
 #warning "Build flags undefined. Please use correct build script"
  ^~~~~~~
  CC(target) Release/obj.target/lmdb-store/dependencies/libmdbx/src/lck-windows.o
  CC(target) Release/obj.target/lmdb-store/dependencies/libmdbx/src/osal.o
  CC(target) Release/obj.target/lmdb-store/dependencies/lz4/lib/lz4.o
  CXX(target) Release/obj.target/lmdb-store/src/node-lmdbx.o
In file included from ../src/node-lmdbx.h:30,
                 from ../src/node-lmdbx.cpp:24:
/home/anurag/.cache/node-gyp/15.2.1/include/node/node.h:855:51: warning: cast between incompatible function types from ‘void (*)(v8::Local<v8::Object>)’ to ‘node::addon_context_register_func’ {aka void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>, void*)’} [-Wcast-function-type]
       (node::addon_context_register_func) (regfunc),                  \
                                                   ^
/home/anurag/.cache/node-gyp/15.2.1/include/node/node.h:873:3: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE_X’
   NODE_MODULE_CONTEXT_AWARE_X(modname, regfunc, NULL, 0)
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/node-lmdbx.cpp:43:5: note: in expansion of macro ‘NODE_MODULE_CONTEXT_AWARE’
     NODE_MODULE_CONTEXT_AWARE(lmdbx, initializeModule)
     ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../src/node-lmdbx.cpp:24:
../src/node-lmdbx.h:130:31: warning: ‘fixedKeySpace’ defined but not used [-Wunused-variable]
 static thread_local KeySpace* fixedKeySpace;
                               ^~~~~~~~~~~~~
../src/node-lmdbx.h:129:30: warning: ‘currentDb’ defined but not used [-Wunused-variable]
 static thread_local DbiWrap* currentDb = nullptr;
                              ^~~~~~~~~
../src/node-lmdbx.h:128:28: warning: ‘lastVersion’ defined but not used [-Wunused-variable]
 static thread_local double lastVersion = 0;
                            ^~~~~~~~~~~
```

OS: Ubuntu

**Comment by patrykwegrzyn** [comment]
Having exactly the same issue on Arch

**Comment by kriszyp** [comment]
Hmm, I think I got it to compile on linux, but now I get an `undefined symbol: mdbx_linux_kernel_version` error.

**Comment by kriszyp** [comment]
I have updated the libmdbx source to the latest "amalgamated" source, and I believe everything looks like it now compiles and runs correctly, at least in basic form. There are some unit tests carried over from the predecessor LMDB project that fail, but that may not affect basic usage.

**Comment by patrykwegrzyn** [comment]
I can confirm it builds `OK `on `Arch Linux`, `hello world` example works too

**Comment by anuragvohraec** [comment]
I was able to run helloworld example now on OS:UBUNTU.

However when I clone this project and tried building it, it gave me a long list of warnings, for the first time. Though the project was build.
When I re-ran `npm install` ,  the problem is not reproducible.
Then I ran test to see every thing OK and following test `npm run test` seems to have failed:

![image](https://user-images.githubusercontent.com/53807480/132597084-26c7f85e-e4fe-4ec7-b5a4-bb03878e2358.png)
